services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/endpoints/S3EndpointResolverContext.java

```diff
@@ -150,6 +177,25 @@ public Builder disableHostPrefixInjection(boolean disableHostPrefixInjection) {
             return this;
         }
 
+        public Builder fipsEnabled(Boolean fipsEnabled) {
+            if (fipsEnabled == null) {
+                this.fipsEnabled = false;
+            } else {
+                this.fipsEnabled = fipsEnabled;
+            }
```

Comment:
We can't ignore null, because `context.fipsEnabled(true); context.fipsEnabled(null);` would be fipsEnabled = true, not false which is what I believe is correct.

If this pattern is weird, I can delay resolving null to false until the context is built instead of doing it in the builder.